### ISSUE TITLE
Embed direct links to HTML5 media in <audio> / <video> tags

### ISSUE THREAD

**Issue by Bugsbane** [diaspora/diaspora#4601]
Currently if a user posts a link to a proprietary video source such as YouTube, the video gets embedded in their post for easy previewing and playing.

If someone posts a link to an open web HTML5 <video> tag compliant video (such as a webm or h264 mp4) it's left as just a link.

The same applies to audio (Soundcloud vs ogg/mp3)

As far as I can see, all this should require is looking for a link starting with http and ending with .ogg/.mp3/.mp4/.webm and if we find it, sticking it in html5 audio or video tags. From the sounds of it Friendica does this already. It would also help Diaspora's core values of encourage decentralization and increasing privacy for our users by removing YouTube / Vimeo / Soundcloud etc in some cases.

The main issue I can see is previews. If this is really an issue, then we could just not bother with them. Users could _additionally_ be allowed to set their own thumbnails in a way similar to posting pictures, eg:

```
+[Keyboard Cat](http://mydomain.com/awesumvids/thumnail2128.jpg)(http://mydomain.com/awesumvids/piano2128.webm)    
```

To embed a video called piano2128.webm with a preview image thumbnail2128.jpg.

Ideally though, people who don't know markup and just post a link would have it "just work"
##

**Comment by cmrd-senya** [comment]
If nobody minds I'd like to work on it.

**Comment by jhass** [comment]
Related discussion https://www.loomio.org/d/LROboQNj/html5-media-embed

**Comment by jhass** [comment]
I think acceptable syntax would be to turn regular links and autolinks into embeds, so basically rendered `<a>` tags if their `href` ends in a matching extension.

**Comment by Bugsbane** [comment]
> If nobody minds I'd like to work on it.

Please be my guest. This seems to be a remarkably simple change for the impact it would have. Real low hanging fruit.

**Comment by svbergerem** [comment]
Another related discussion: http://talk.commonmark.org/t/embedded-audio-and-video/441

I think it would be great if we could implement this as a markdown-it plugin. There might be also other projects who are interested in this plugin. In the commonmark thread there is already one dev who started writing a markdown-it plugin. Maybe we can join forces.

Here are some things we should keep in mind / think about;
* Should we have a configurable list of sources or do we accept arbitrary sources?
* It would be great if the user has to activate the player before we load any files.
* What kind of information do we present our users before they activate the player?

**Comment by Flaburgan** [comment]
>I think it would be great if we could implement this as a markdown-it plugin

+1

**Comment by v3ss0n** [comment]
Author of example mulit-media plugin from that post. I am buidling markdown-it plugin for it. Will be avaliable within 48 hrs.

**Comment by cmrd-senya** [comment]
Voting for the embedding syntax is [in progress](https://www.loomio.org/d/LROboQNj/html5-media-embed).

**Comment by v3ss0n** [comment]
for CommonMark ? or Diaspora ? 

IMG Syntax directly relates to Video/Audio (media) , Link syntax is out of scope for displaying video/audio/images .

**Comment by cmrd-senya** [comment]
For Diaspora.

It is not stated anywhere though.

**Comment by gaffatape** [comment]
I think the approach need to be based on the normal principles of OpenGraph.

More detail in my comment on the other related issue:
https://github.com/diaspora/diaspora/pull/6418#issuecomment-162530861

**Comment by Bugsbane** [comment]
We should definitely accept arbitrary sources, as the whole point is to help decentralize media on the web with open standards, not to just include an arbitrary handful of new gatekeepers that we accept.

I do agree though that we should protect users privacy by not having any automatic per u user calls without the users request. It may be useful though to have the pod cache the filename/server/metadata to present to the user to help them decide if they want to play the Duke or not. For further privacy gains, this info could be federated the same as post text, so only the original posters pod loads the file metadata without the user pushing play (which is reasonable for the user to assume as they're paying that file).

I would suggest that the best info to share would be the meta title of available and the file name of not, along with the hosting server name and creation date. This is the most equivalent to what users are used to with video hosting platforms like YouTube and Vimeo etc

**Comment by Flaburgan** [comment]
will be fixed by #6418

**Comment by SuperTux88** [comment]
Fixed by #6418
